File: Demo_OC/LCDownloadDemo/ViewController.m
//
//  ViewController.m
//  LCDownloadDemo
//
//  Created by LiuChang on 16-6-24.
//  Copyright (c) 2016年 LiuChang. All rights reserved.
//

#import "ViewController.h"
#import "LCDownloadManager.h"

static const NSUInteger Tag1 = 10;
static const NSUInteger Tag2 = 1007;
NSString *URL = @"http://baobab.wdjcdn.com/1455782903700jy.mp4";
@interface ViewController ()
{
    UILabel *_label1;
    UILabel *_label2;
    UIButton *_button1;
    UIButton *_button2;
    UIProgressView *_pv1;
    UIProgressView *_pv2;
}
@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    LCDownloadManager *lc_D = [[LCDownloadManager alloc]init];
    
    UILabel *label1 = [[UILabel alloc]initWithFrame:CGRectMake(0, 30, 50, 50)];
    label1.text = [NSString stringWithFormat:@"%.2f",[lc_D progressWithTag:Tag1]];
    label1.textColor = [UIColor blackColor];
    [self.view addSubview:label1];
    _label1 = label1;
    UIProgressView *pv1 = [[UIProgressView alloc]initWithFrame:CGRectMake(CGRectGetMaxX(label1.frame), 50, 200, 30)];
    pv1.progressTintColor = [UIColor blueColor];
    pv1.progress = [lc_D progressWithTag:Tag1];
    [self.view addSubview:pv1];
    _pv1 = pv1;
    UIButton *button1 = [[UIButton alloc]init];
    button1.frame = CGRectMake(CGRectGetMaxX(pv1.frame), 30, 50, 50);
    [button1 setImage:[UIImage imageNamed:@"start"] forState:UIControlStateNormal];
    [button1 setImage:[UIImage imageNamed:@"pauce"] forState:UIControlStateSelected];
    [button1 addTarget:self action:@selector(click1:) forControlEvents:(UIControlEventTouchUpInside)];
    _button1 = button1;
    [self.view addSubview:button1];
    
    UIButton *rBtn = [[UIButton alloc]init];
    rBtn.frame = CGRectMake(CGRectGetMaxX(button1.frame), 30, 50, 50);
    [rBtn setImage:[UIImage imageNamed:@"delete"] forState:UIControlStateNormal];
    [self.view addSubview:rBtn];
    [rBtn addTarget:self action:@selector(remove1:) forControlEvents:(UIControlEventTouchUpInside)];
    
    
    UILabel *label2 = [[UILabel alloc]initWithFrame:CGRectMake(0, 100, 50, 50)];
    label2.text = [NSString stringWithFormat:@"%.2f",[lc_D progressWithTag:Tag2]];
    label2.textColor = [UIColor blackColor];
    [self.view addSubview:label2];
    _label2 = label2;
    UIProgressView *pv2 = [[UIProgressView alloc]initWithFrame:CGRectMake(CGRectGetMaxX(label1.frame), 120, 200, 30)];
    pv2.progressTintColor = [UIColor blueColor];
    pv2.progress = [lc_D progressWithTag:Tag2];
    [self.view addSubview:pv2];
    _pv2= pv2;
    UIButton *button2 = [[UIButton alloc]init];
    button2.frame = CGRectMake(CGRectGetMaxX(pv2.frame), 100, 50, 50);
    [button2 setImage:[UIImage imageNamed:@"start"] forState:UIControlStateNormal];
    [button2 setImage:[UIImage imageNamed:@"pauce"] forState:UIControlStateSelected];
    [button2 addTarget:self action:@selector(click2:) forControlEvents:(UIControlEventTouchUpInside)];
    _button2 = button2;
    [self.view addSubview:button2];
    
    UIButton *rBtn2 = [[UIButton alloc]init];
    rBtn2.frame = CGRectMake(CGRectGetMaxX(button2.frame), 100, 50, 50);
    [rBtn2 setImage:[UIImage imageNamed:@"delete"] forState:UIControlStateNormal];
    [self.view addSubview:rBtn2];
    [rBtn2 addTarget:self action:@selector(remove2:) forControlEvents:(UIControlEventTouchUpInside)];
   
    
    UIButton *rBtn3 = [[UIButton alloc]init];
    rBtn3.backgroundColor = [UIColor orangeColor];
    rBtn3.frame = CGRectMake(200, 200, 50, 30);
    [rBtn3 setTitle:@"清空" forState:(UIControlStateNormal)];
    [self.view addSubview:rBtn3];
    [rBtn3 addTarget:self action:@selector(removeAll:) forControlEvents:(UIControlEventTouchUpInside)];
}
- (NSString *)getText:(CGFloat)progress {
    NSString *scaleStr = [NSString stringWithFormat:@"%.2f",progress];
    return scaleStr;
}

- (void)removeAll:(UIButton *)button {
    _label1.text = @"0.00";
    _pv1.progress = 0;
    _label2.text = @"0.00";
    _pv2.progress = 0;
    _button1.selected = NO;
    _button2.selected = NO;
    [[LCDownloadManager sharedInstance] removeAllFileData];
}

- (void)click1:(UIButton *)button {
    button.selected = !button.selected;
    [[LCDownloadManager sharedInstance] downloadDataWithURL:URL tag:Tag1 resume:button.selected progress:^(CGFloat progress) {
        dispatch_async(dispatch_get_main_queue(), ^{
            _label1.text = [self getText:progress];
            _pv1.progress = progress;
            
        });
    } state:nil];

}

- (void)click2:(UIButton *)button {
    button.selected = !button.selected;
    [[LCDownloadManager sharedInstance] downloadDataWithURL:URL tag:Tag2 resume:button.selected progress:^(CGFloat progress) {
        dispatch_async(dispatch_get_main_queue(), ^{
            _label2.text = [self getText:progress];
            _pv2.progress = progress;
            
        });
    } state:nil];

}

- (void)remove1:(UIButton *)button {
    _label1.text = @"0.00";
    _pv1.progress = 0;
    _button1.selected = NO;
    [[LCDownloadManager sharedInstance] removeFileDataWithTag:Tag1];
}
-(void)remove2:(UIButton *)button {
    _label2.text = @"0.00";
    _pv2.progress = 0;
    _button2.selected = NO;
    [[LCDownloadManager sharedInstance] removeFileDataWithTag:Tag2];
}
- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

@end

File: DemoSwift/DemoSwift/ViewController.swift
//
//  ViewController.swift
//  LCDownloadDemo
//
//  Created by 刘畅 on 16/6/24.
//  Copyright © 2016年 ifdoo. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    
    let Tag1 = 12334
    let Tag2 = 980
    let url = "http://118.31.132.13:8002/images/f/f3/CCP_REST_DEMO_Objective-C_v2.6r.zip"
    
    weak var pv1: UIProgressView!
    weak var pv2: UIProgressView!
    weak var label1: UILabel!
    weak var label2: UILabel!
    weak var button1: UIButton!
    weak var button2: UIButton!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        let button = UIButton()
        let sd = LCSwiftDownload.sharedInstance

        
       
        let p1 = sd.getProgressValue(tag: Tag1)
        let pv1 = UIProgressView()
        let label1 = UILabel()
        label1.text = String(format: "%.2f", p1)
        label1.frame = CGRect(x: 0, y: 50, width: 50, height: 50)
        label1.textColor = UIColor.black
        self.view.addSubview(label1)
        self.label1 = label1
        pv1.progress = Float(p1)
        pv1.frame = CGRect(x: 50, y: 70, width: 200, height: 10)
        pv1.progressTintColor = UIColor.blue
        self.view.addSubview(pv1)
        self.pv1 = pv1
        button.frame = CGRect(x: pv1.frame.maxX, y: 50, width: 50, height: 50)
        button.setTitleColor(UIColor.black, for: .normal)
        button.addTarget(self, action: #selector(self.click(button:)), for: .touchUpInside)
        button.setImage(UIImage(named: "start"), for: .normal)
        button.setImage(UIImage(named: "pauce"), for: .selected)
        self.view.addSubview(button)
        self.button1 = button
        
        let removeButton = UIButton()
        removeButton.frame = CGRect(x: button.frame.maxX, y: 50, width: 50, height: 50)
        removeButton.setTitleColor(UIColor.black, for: .normal)
        removeButton.addTarget(self, action: #selector(self.remove(button:)), for: .touchUpInside)
        removeButton.setImage(UIImage(named: "delete"), for: .normal)
        self.view.addSubview(removeButton)
       
        
        let p2 = sd.getProgressValue(tag: Tag2)
        
        let label2 = UILabel()
        label2.text = String(format: "%.2f", p2)
        label2.frame = CGRect(x: 0, y: 140, width: 50, height: 50)
        label2.textColor = UIColor.black
        self.view.addSubview(label2)
        self.label2 = label2
        
        let pv2 = UIProgressView()
        pv2.progress = Float(p2)
        pv2.progressTintColor = UIColor.blue
        pv2.frame = CGRect(x: 50, y: 160, width: 200, height: 10)
        self.view.addSubview(pv2)
        self.pv2 = pv2
        let button1 = UIButton()
        button1.frame = CGRect(x: pv1.frame.maxX, y: 140, width: 50, height: 50)
        button1.setTitleColor(UIColor.black, for: .normal)
        button1.addTarget(self, action: #selector(self.click1(button:)), for: .touchUpInside)
        button1.setImage(UIImage(named: "start"), for: .normal)
        button1.setImage(UIImage(named: "pauce"), for: .selected)
        self.view.addSubview(button1)
        self.button2 = button1
        let rbutton1 = UIButton()
        rbutton1.frame = CGRect(x: button1.frame.maxX, y: 140, width: 50, height: 50)
        rbutton1.setTitleColor(UIColor.black, for: .normal)
        rbutton1.addTarget(self, action: #selector(self.remove1(button:)), for: .touchUpInside)
        rbutton1.setImage(UIImage(named: "delete"), for: .normal)
        self.view.addSubview(rbutton1)
        
      
        
    }
    
    @objc func click(button: UIButton) {
        button.isSelected = !button.isSelected
    let sd = LCSwiftDownload.sharedInstance
        sd.downloadData(url: url, tag: Tag1, resume: button.isSelected, progerss: { (progerssValue) in
            DispatchQueue.main.async {
                self.label1.text = String(format: "%.2f", progerssValue)
                self.pv1.progress = progerssValue

            }
            }) { (state) in
                
        }
    
    }
    
    @objc func remove(button: UIButton) {
        pv1.progress = 0
        label1.text = "0.00"
        self.button1.isSelected = false
        LCSwiftDownload.sharedInstance.removeFileDownloadedData(tag: Tag1)
    }
    
    @objc func click1(button: UIButton) {
        button.isSelected = !button.isSelected
        let sd = LCSwiftDownload.sharedInstance
        sd.downloadData(url: url, tag: Tag2, resume: button.isSelected, progerss: { (progerssValue) in
            self.label2.text = String(format: "%.2f", progerssValue)
            self.pv2.progress = progerssValue
            }, state: nil)
      
    }
    
    @objc func remove1(button: UIButton) {
        pv2.progress = 0
        label2.text = "0.00"
        self.button2.isSelected = false 
        LCSwiftDownload.sharedInstance.removeFileDownloadedData(tag: Tag2)
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}

